fundingmanager.go

```diff
@@ -1190,6 +1191,19 @@ func (f *fundingManager) handleFundingOpen(fmsg *fundingOpenMsg) {
 		return
 	}
 
+	shutdown, err := getUpfrontShutdownScript(fmsg.peer,
```

Comment:
Could we do this on a per-channel basis? Meaning that we accept a upfront address when we open a channel. That way we can actually expose it on the RPC (in a later PR).

Reply:
The problem with requiring user input for upfront shutdown is that they aren't around to signal whether we should use upfront shutdown addresses in our `acceptChannel` messages when peers initiate channel opens.

Re getting addresses from users, I was thinking that we have this flag (which will enable for accepts and opens) and then allow users to specify an upfront address via lncli/rpc when opening channels. This would provide the ability to set a specific address when `enable_upfront_shutdown` is true, and to signal upfront shutdown for the specific channel that they're opening if `enable_upfront_shutdown` is false.

This does require some knowledge of peer feature bits, but I thinks something of the sort may be in the pipeline?